File: Domain/Entities/Notification.cs
using Domain.Common;

namespace Domain.Entities;

public class Notification
{
    public Guid NotificationId { get; set; }
    public Guid UserId { get; set; }
    public string SubjectName { get; set; }
    public string NotificationType { get; set; }
    public DateTime DateSent { get; set; }
    public bool ReadStatus { get; set; }
    public Notification()
    {

    }
    public Notification(Guid userId, string subjectName, string notificationType)
    {
        NotificationId = Guid.NewGuid();
        SubjectName = subjectName;
        NotificationType = notificationType;
        UserId = userId;
        DateSent = DateTime.UtcNow;
        ReadStatus = false;
    }
    public Result<Notification> UpdateReadStatus()
    {
        ReadStatus = true;
        return Result<Notification>.Success(this);
    }
}

File: UI/steam-frontend/src/app/core/models/notification/notification.ts
export interface INotification {
    notificationId: string,
    userId: string,
    subjectName: string,
    notificationType: string,
    dateSent?: Date,
    readStatus: boolean
}

export interface INotificationPaginated {
    notifications: INotification[],
    pageIndex: number,
    totalPages: number,
    totalItems: number
}